features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/SessionReplayConfigurationBuilderTest.kt

```diff
@@ -230,4 +239,186 @@ internal class SessionReplayConfigurationBuilderTest {
+        whenever(mockSecondExtension.name()).thenReturn("secondExtension")
+
+        // When
+        testedBuilder.addExtensionSupport(mockFirstExtension)
+        testedBuilder.addExtensionSupport(mockSecondExtension)
+        val config = testedBuilder.build()
+
+        // Then
+        assertThat(config.customMappers).containsAll(
+            setOf(fakeTextMapperTypeWrapper, fakeImageMapperTypeWrapper)
+        )
+    }
+
+    @Test
+    fun `M take first value W addExtensionSupport { duplicate mappers }`(
+        @Mock mockFirstExtension: ExtensionSupport,
+        @Mock mockSecondExtension: ExtensionSupport,
+        @Mock mockMapper: WireframeMapper<ImageView>
+    ) {
+        // Given
+        val fakeMapperTypeWrapper = MapperTypeWrapper(
+            ImageView::class.java,
+            mockMapper
+        )
+
+        whenever(mockFirstExtension.getCustomViewMappers())
+            .thenReturn(listOf(fakeMapperTypeWrapper))
+
+        whenever(mockSecondExtension.getCustomViewMappers())
+            .thenReturn(listOf(fakeMapperTypeWrapper))
+
+        // When
+        testedBuilder.addExtensionSupport(mockFirstExtension)
+        testedBuilder.addExtensionSupport(mockSecondExtension)
+        val config = testedBuilder.build()
+
+        // Then
+        assertThat(config.customMappers).isEqualTo(mockFirstExtension.getCustomViewMappers())
+    }
+
+    @Test
+    fun `M warn W addExtensionSupport { duplicate mappers }`(
+        @Mock mockFirstExtension: ExtensionSupport,
+        @Mock mockSecondExtension: ExtensionSupport,
+        @Mock mockMapper: WireframeMapper<ImageView>
+    ) {
+        // Given
+        val fakeMapperTypeWrapper = MapperTypeWrapper(
+            ImageView::class.java,
+            mockMapper
+        )
+
+        whenever(mockFirstExtension.getCustomViewMappers())
+            .thenReturn(listOf(fakeMapperTypeWrapper))
+        whenever(mockFirstExtension.name()).thenReturn("firstExtension")
+        whenever(mockSecondExtension.name()).thenReturn("secondExtension")
+
+        whenever(mockSecondExtension.getCustomViewMappers())
+            .thenReturn(listOf(fakeMapperTypeWrapper))
+
+        // When
+
+        testedBuilder.addExtensionSupport(mockFirstExtension)
+        testedBuilder.addExtensionSupport(mockSecondExtension)
+
+        val output = capturePrintOutput {
+            testedBuilder.build()
+        }
+
+        val expected = String.format(DUPLICATE_MAPPER_DETECTED, ImageView::class.java)
+
+        // Then
+        assertThat(output.trim()).isEqualTo(expected)
+    }
+
+    @Test
+    fun `M warn W addExtensionSupport { duplicate extensions }`(
+        @Mock mockExtension: ExtensionSupport
+    ) {
+        // When
+        val output = capturePrintOutput {
+            testedBuilder.addExtensionSupport(mockExtension)
+            testedBuilder.addExtensionSupport(mockExtension)
+        }
+
+        // Then
+        assertThat(output.trim()).isEqualTo(DUPLICATE_EXTENSION_DETECTED)
```

Comment:
Suggestion:
In this unit test you can have two different mock extension with the same name, and then verify also that the the configuration contains only the first extension adde.